web/models.py

```
@@ -3046,3 +3046,85 @@
+class VirtualClassroomParticipant(models.Model):
+    """Model for tracking active participants in a virtual classroom."""
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    classroom = models.ForeignKey(VirtualClassroom, on_delete=models.CASCADE)
+    joined_at = models.DateTimeField(auto_now_add=True)
+    last_active = models.DateTimeField(auto_now=True)
+    seat_id = models.CharField(max_length=20, null=True, blank=True)
+
+    class Meta:
+        unique_together = ('classroom', 'user')
+
+    def __str__(self):
```

Ruff (0.11.9)

Missing return type annotation for special method `__str__`

Add return type annotation: `str`

(ANN204)